src/gui/mainwindow.cpp

```diff
@@ -1095,14 +1102,32 @@ void MainWindow::dropEvent(QDropEvent *event)
         files = event->mimeData()->text().split('\n');
     }
 
-    // Add file to download list
+    // differentiate ".torrent" files and others
+    QStringList otherFiles;
+    QStringList torrentFiles;
+    foreach (const QString &s, files) {
+        if (s.endsWith(".torrent", Qt::CaseInsensitive))
+            torrentFiles << s;
+        else
+            otherFiles << s;
+    }
+
+    // Download torrents
     const bool useTorrentAdditionDialog = AddNewTorrentDialog::isEnabled();
-    foreach (QString file, files) {
-        qDebug("Dropped file %s on download list", qPrintable(file));
-        if (useTorrentAdditionDialog)
-            AddNewTorrentDialog::show(file, this);
+    foreach (const QString &t, torrentFiles) {
```

Comment:
Please do not overuse the meaningless names. `torrentFile` (or just `file`) is better.